### PR TITLE
Initial preparation for 0.4.0 tag

### PR DESCRIPTION
**Pull request by alanz**
Still needs the editorial part of the ChangeLog

### CODE REVIEW

**Review by fendor — APPROVED** [review]
Nice, LGTM!

File: ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,105 @@
...
+- tech(nix): update niv and remove allowbroken
+([#350](https://github.com/haskell/haskell-language-server/pull/350) by @willbush)
+- Update VS Code Haskell URL/repo
+([#338](https://github.com/haskell/haskell-language-server/pull/338) by @Sir4ur0n)
+- doc(hack): Add explanation to hack and test HLS
+([#329](https://github.com/haskell/haskell-language-server/pull/329) by @Sir4ur0n)
+- Apply the module pragmas for evaluation
+([#322](https://github.com/haskell/haskell-language-server/pull/322) by @pepeiborra)
+- Use hie-bios implicit cradle with a fallback to the cabal-helper one
+([#110](https://github.com/haskell/haskell-language-server/pull/110) by @jneira)
```
Comment (jneira) [inline ChangeLog.md:79]: This one was closed without merging it

**Review by jneira — APPROVED** [review]
LGTM

**Review by Ailrun — APPROVED** [review]
Looks good!

**Pull request merged**